src/core/infer_request.h

```diff
@@ -379,8 +380,15 @@ class InferenceRequest {
   // Create a copy of 'from' suitable for use as a "null" request as
   // required for the direct sequence batcher. The returned copy will
   // contain only the minimum content required for a null request.
+  // The statistics of the copy will not be collected.
   static InferenceRequest* CopyAsNull(const InferenceRequest& from);
 
+  // Create a copy of 'from' suitable for use. The caller must be aware that
```

Comment:
I don't like the semantics of this copy. Can you just regenerate the request again when it is needed multiple times for warmup. Also there is no need to suppress stats collection for warmup.